src/cascade/input_data/configuration/construct_bundle.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+import pandas as pd
+
+from cascade.input_data import InputDataError
+from cascade.input_data.configuration.construct_study import \
+    get_bundle_study_covariates
+from cascade.input_data.configuration.id_map import make_integrand_map
+from cascade.input_data.db.bundle import _get_bundle_id, _get_bundle_data
+from cascade.dismod.db.metadata import DensityEnum
+from cascade.core import getLoggers
+
+CODELOG, MATHLOG = getLoggers(__name__)
+
+
+def _normalize_measures(data):
+    """Transform measure_ids into canonical measure names, for instance,
+    GBD measure 38 for birth prevalence becomes pini.
+    """
+    data = data.copy()
+    gbd_measure_id_to_integrand = make_integrand_map()
+    if any(data.measure_id == 6):
+        MATHLOG.warn(f"Found incidence, measure_id=6, in data. Should be Tincidence or Sincidence.")
+    try:
+        data["measure"] = data.measure_id.apply(lambda k: gbd_measure_id_to_integrand[k].name)
+    except KeyError as ke:
+        raise RuntimeError(
+            f"The bundle data uses measure {str(ke)} which doesn't map "
+            f"to an integrand. The map is {gbd_measure_id_to_integrand}."
+        )
+    return data
+
+
+def _normalize_sex(data):
+    """Transform sex_ids from 1, 2, 3 to male, female, both.
+    """
+    data = data.copy()
+    try:
+        data["sex"] = data.sex_id.apply({1: "Male", 2: "Female", 3: "Both", 4: "Unspecified"}.get)
+    except KeyError as ke:
+        raise InputDataError(f"Unrecognized sex id") from ke
+    return data
+
+
+def _normalize_bundle_data(data):
+    """Normalize bundle columns, strip extra columns and index on `seq`
+    """
+    data = _normalize_measures(data)
+    data = _normalize_sex(data)
+
+    data = data.set_index("seq")
+
+    cols = ["measure", "mean", "sex", "sex_id", "standard_error", "age_start", "age_end", "year_start", "year_end", "location_id"]
+
+    return data[cols]
+
+
+def bundle_to_observations(config, bundle_df):
+    """
+    Convert bundle into an internal format. It removes the sex column and changes
+    location to node. It also adjusts for the demographic specification.
+    """
+    if "location_id" in bundle_df.columns:
+        location_id = bundle_df["location_id"]
+    else:
+        location_id = np.full(len(bundle_df), config.location_id, dtype=np.int)
+
+    # assume using demographic notation because this bundle uses it.
+    demographic_interval_specification = 0
```

Comment:
Can it be a 1 somewhere else, or are we planning to add a function call here?

Reply:
This is a bit of behavior that the modelers need to fix in order for the code to work correctly. This is a little yell that it's wrong.

Reply:
It's the number of years to add. Maybe I could change the variable name? This part of the code is a terrible FIXME moment b/c we don't trust the data at all. This should be reading a column of True or False and then adding 1 or 0.

Reply:
Oh, that was not clear to me. I think a different name would help `demographic_interval_offset` or something maybe.